sunpy/lightcurve/sources/sit.py

```diff
@@ -0,0 +1,297 @@
+    #     figure.show()
+
+    #     return figure
+
+    @classmethod
+    def _get_default_url(cls):
+        """
+           Retrieve latest SIT data for 1 minute average of 4He specie from ahead spacecraft 
+           ( if no other data is specified )
+        """
+        now = datetime.datetime.utcnow()
+        time_range = TimeRange(datetime.datetime(now.year, now.month, now.day), now)
+        url_does_exist = net.url_exists(cls._get_url_for_date_range(time_range))
+        while not url_does_exist:
+            time_range = TimeRange(time_range.start-datetime.timedelta(days=1),
+                                   time_range.start)
+            url_does_exist = net.url_exists(cls._get_url_for_date_range(time_range))
+        return cls._get_url_for_date_range(time_range, "ahead", '1min', "4He")
+
+    
+
+    @staticmethod
+    def _get_url_for_date_range(*args):
+        """Returns a list of URLs to the SIT data for the specified range of dates along with the 
+         tuple containing specified start and end dates .
+        Parameters
+        ----------
+        args : TimeRange, datetimes, date strings
+            Date range should be specified using a TimeRange, or start
+            and end dates at datetime instances or date strings.
+        stereo_spacecraft : string
+            SIT stereo spacecraft (default = "ahead")
+        atomic_specie : string 
+            atomic specie to consider ( default = "4He")
+        type_of_average : string
+            average type to retrieve (default = "1min")
+        """
+        # # If time duration given as sunpy.time.TimeRange
+        if len(args) == 4 and isinstance(args[0], TimeRange):
+            start = args[0].start
+            end = args[0].end
+            stereo_spacecraft = args[1]
+            type_of_average = args[2]
+            atomic_specie = args[3]
+
+        elif len(args) == 5:
+            #If start and end dates for observation period is given
+            start = parse_time(args[0])
+            end = parse_time(args[1])
+            stereo_spacecraft = args[2]
+            type_of_average = args[3]
+            atomic_specie = args[4]
+
+        if end < start:
+            raise ValueError('start time > end time')
+
+
+        # find out base url of data to query from the stereo spacecraft, average type and atomic specie quried for
+        base_url = 'http://www.srl.caltech.edu/STEREO/DATA/SIT/' + stereo_spacecraft + '/' + type_of_average + '/' 
+
+        if type_of_average == '1min' or type_of_average == '10min':
+            base_url = base_url + atomic_specie + '/'
+        
+        #adding the file name to base url
+        base_url = base_url + 'SIT_' + stereo_spacecraft.capitalize() + '_' + type_of_average + '_' + atomic_specie + '_'
+
+        #Date Generator to generate dates in between the start and end dates. Inclusive of both end and start dates.         
+        def daterange(start_date, end_date, delta = 'day'):
```

Comment:
The easiest way of doing this would probably be a combination of `datetime.timedelta` and `TimeRange` i.e. for days:

```python
tr = TimeRange('2012/1/1T00:00:00','2012/1/5T00:00:00')

days = list()
for day in range(int(np.ceil(tr.days.value))):
    days.append(tr.start.replace(day=tr.start.day+day))
```

and you should be able to do s/day/month for months.